static/js/components/PostList.js

```diff
@@ -5,50 +5,56 @@ import CompactPostDisplay from "./CompactPostDisplay"
 
 import type { Post } from "../flow/discussionTypes"
 
-type PostListProps = {
+type Props = {
   posts: Array<Post>,
-  showChannelLinks?: boolean,
-  showPinUI?: boolean,
+  showChannelLinks: boolean,
+  showPinUI: boolean,
   toggleUpvote: Post => void,
   togglePinPost?: Post => Promise<*>,
-  isModerator?: boolean,
+  isModerator: boolean,
   reportPost: (p: Post) => void,
   removePost?: (p: Post) => void
 }
 
-const PostList = (props: PostListProps) => {
-  const {
-    posts,
-    showChannelLinks,
-    showPinUI,
-    isModerator,
-    toggleUpvote,
-    togglePinPost,
-    reportPost,
-    removePost
-  } = props
+export default class PostList extends React.Component<Props> {
```

Comment:
Do functions as react components not work for flow typing purposes?

Reply:
Oh yeah, they do just fine - I refactored this component into a class in order to be able to set `defaultProps`